bigtable-dataflow-parent/bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableScanConfiguration.java

```diff
@@ -177,8 +180,12 @@ public CloudBigtableScanConfiguration build() {
         if (scan == null) {
           scan = new Scan();
         }
-        ReadRowsRequest.Builder builder = Adapters.SCAN_ADAPTER.adapt(scan, readHooks);
-        request = readHooks.applyPreSendHook(builder.build());
+        Query query = Query.create(tableId);
+        Adapters.SCAN_ADAPTER.adapt(scan, readHooks, query);
+        RequestContext requestContext =
+            RequestContext.create(InstanceName.of(projectId, instanceId), "");
+        readHooks.applyPreSendHook(query);
```

Comment:
`appProfileId` is not available at this class... I added an empty string to for the same in `RequestContext.create()` constructor. should this be fine?

Reply:
You should be able to get the appProfileId from the `additionalConfiguration`.

Reply:
The reason I thought `additionConfiguration` might not have `appProfileId` because `dataflow.CloudBigtableScanConfiguration` doesn't have  **`withAppProfileId()`**, but it's present  in `beam.CloudBigtableScanConfiguration`.

Would it be available through `withConfiguration(key, value)`?